src/iosrv/init.c

```diff
@@ -232,6 +236,66 @@ dss_topo_init()
 	dss_tgt_nr = dss_tgt_nr_get(dss_core_nr, nr_threads);
 }
 
+/* daos_server dRPC method notifyReady (see also notifyReady) */
+#define SRV_NOTIFY_READY 0
+
+/* Notify daos_server that we are ready (e.g., to receive dRPC requests). */
+static int
+notify_ready(void)
+{
+	Drpc__Call	req = DRPC__CALL__INIT;
+	Drpc__Response *resp;
+	int		rc;
+
+	req.module = DRPC_MODULE_SRV;
+	req.method = SRV_NOTIFY_READY;
+
+	rc = drpc_call(dss_drpc_ctx, R_SYNC, &req, &resp);
+	if (rc != 0)
+		return rc;
+
+	drpc__response__free_unpacked(resp, NULL);
+	return rc;
+}
+
+static int
+drpc_init(void)
+{
+	char   *path;
+	int	rc;
+
+	rc = asprintf(&path, "%s/%s", dss_socket_dir, "daos_server.sock");
+	if (rc < 0) {
+		rc = -DER_NOMEM;
+		goto out;
+	}
+
+	dss_drpc_ctx = drpc_connect(path);
+	if (dss_drpc_ctx == NULL) {
+		rc = -DER_NOMEM;
+		goto out_path;
+	}
+
+	rc = notify_ready();
+	if (rc != 0)
+		drpc_close(dss_drpc_ctx);
```

Comment:
Should the ctx be reset to NULL after this?

Reply:
Yes, will fix.